application-task-default/src/main/java/com/xwiki/task/internal/DateFormatOverrideListener.java

```diff
@@ -0,0 +1,119 @@
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package com.xwiki.task.internal;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Provider;
+import javax.inject.Singleton;
+
+import org.xwiki.component.annotation.Component;
+import org.xwiki.component.phase.Initializable;
+import org.xwiki.component.phase.InitializationException;
+import org.xwiki.configuration.ConfigurationSaveException;
+import org.xwiki.configuration.ConfigurationSource;
+import org.xwiki.observation.AbstractEventListener;
+import org.xwiki.observation.event.Event;
+
+/**
+ * Listener that assures the compatibility with older versions that upgrade. Migrates the date format preferences
+ * from the task application configuration to the new Date macro configuration.
+ *
+ * @version $Id$
+ * @since 3.0
+ */
+@Component
+@Named(DateFormatOverrideListener.ROLE_HINT)
+@Singleton
+public class DateFormatOverrideListener extends AbstractEventListener implements Initializable
+{
+    /**
+     * The role hint.
+     */
+    public static final String ROLE_HINT = "com.xwiki.task.internal.DateFormatOverrideListener";
+
+    private static final String DATE_STORAGE_FORMAT_KEY = "storageDateFormat";
+
+    private static final String DATE_DISPLAY_FORMAT_KEY = "displayDateFormat";
+
+    @Inject
+    @Named("taskmanager")
+    private Provider<ConfigurationSource> configurationSourceProvider;
+
+    @Inject
+    @Named("datemacro")
+    private Provider<ConfigurationSource> dateMacroConfigurationSourceProvider;
+
+    /**
+     * Default constructor.
+     */
+    public DateFormatOverrideListener()
+    {
+        super(ROLE_HINT, Collections.emptyList());
+    }
+
+    @Override
+    public void initialize() throws InitializationException
+    {
+        // When this component is initialized (on instance start up / new version install) it should replace the
+        // DocumentInstanceOutputFilterStream from the component manager.
+        try {
+            maybeReplaceDateFormat();
+        } catch (ConfigurationSaveException e) {
+            throw new InitializationException(e.getMessage());
```

Comment:
I think it would be better to ignore it rather than throw an InitializationException. I believe the thrown exception will print the entire stack in the server logs which is quite the overkill. I believe the exception is thrown only when restarting the wiki, right? Or is it thrown even at upgrade time as well

Reply:
I'm not entirely sure what you're referring to. The initialize method is called at every startup and every upgrade, so an error may occur at every start and upgrade. I think it's ok to log the error, as it should not be a common occurrence.